input/kafkamdm/lag_monitor.go

```diff
@@ -50,71 +65,45 @@ func (l *lagLogger) Min() int {
 	}
 	min := -1
 	for _, m := range l.measurements {
-		if min < 0 || m < min {
-			min = m
+		lag := int(m.highWaterMark - m.readOffset)
+		if min < 0 || lag < min {
```

Comment:
tiny optimization: at this point we know that `len(l.measurements) > 0` so we could initialize `min` with the lag of the first measurement and then iterate over `l.measurements[1:]`. This would save one loop and this `if` condition would be simplified because it only needs to be `lag < min` instead of `min < 0 || lag < min`.